web/src/echo.js

```diff
@@ -210,35 +209,14 @@ export function is_slash_command(content) {
 }
 
 export function try_deliver_locally(message_request, insert_new_messages) {
-    if (message_lists.current === undefined) {
-        return undefined;
-    }
-
     if (markdown.contains_backend_only_syntax(message_request.content)) {
         return undefined;
     }
 
-    if (narrow_state.filter() !== undefined && !narrow_state.filter().can_apply_locally(true)) {
-        return undefined;
-    }
-
     if (is_slash_command(message_request.content)) {
         return undefined;
     }
 
-    if (!message_lists.current.data.fetch_status.has_found_newest()) {
-        // If the current message list doesn't yet have the latest
-        // messages before the one we just sent, local echo would make
-        // it appear as though there were no messages between what we
-        // have and the new message we just sent, when in fact we're
-        // in the process of fetching those from the server.  In this
-        // case, it's correct to skip local echo; we'll get the
-        // message we just sent placed appropriately when we get it
-        // from either server_events or message_fetch.
-        blueslip.info("Skipping local echo until newest messages get loaded.");
-        return undefined;
-    }
```

Comment:
Hmm, I'm pretty confident that this logic is important in some situations -- like if you narrow to a long topic `near:1` and then receive a message to that topic.

Reply:
Yeah, it is important but doesn't actually work if the message was sent outside of the current message list. Also,
the path through which we add local echo messages to the message list also has the same check which runs for `message_lists.all_rendered_message_lists()`.

```js
export function add_new_messages(
    messages: Message[],
    msg_list: MessageList,
): RenderInfo | undefined {
    if (!msg_list.data.fetch_status.has_found_newest()) {
        // We don't render newly received messages for the message list,
        // if we haven't found the latest messages to be displayed in the
        // narrow. Otherwise the new message would be rendered just after
        // the previously fetched messages when that's inaccurate.
        msg_list.data.fetch_status.update_expected_max_message_id(messages);
        return undefined;
    }
    return add_messages(messages, msg_list, {messages_are_new: true});
}
```